TYPO3.Neos/Tests/Unit/TypoScript/ConvertUrisImplementationTest.php

```diff
@@ -301,4 +301,31 @@ public function evaluateReplaceResourceLinkTargets()
         $actualResult = $this->convertUrisImplementation->evaluate();
         $this->assertSame($expectedResult, $actualResult);
     }
+
+    /**
+     * This test checks that targets for node links are correctly replaced to their html links when evaluated from foreign format like json or xml.sitemap
+     *
+     * @test
+     */
+    public function evaluateReplaceNodeUrisFromForeignFormat()
+    {
+        $nodeIdentifier = 'aeabe76a-551a-495f-a324-ad9a86b2aff7';
+        $value = 'This string contains a node link and an external link: The node link <a href="node://' . $nodeIdentifier . '">example</a> and an external link <a href="http://www.example.org">example3</a>';
+        $this->addValueExpectation($value);
+
+        $this->mockWorkspace->expects($this->any())->method('getName')->will($this->returnValue('live'));
+        $this->mockActionRequest->expects($this->any())->method('getFormat')->will($this->returnValue('json.something'));
+
+        $self = $this;
+        $this->mockLinkingService->expects($this->atLeastOnce())->method('resolveNodeUri')->will($this->returnCallback(function ($nodeUri) use ($self, $nodeIdentifier) {
+            if ($nodeUri !== 'node://' . $nodeIdentifier) {
+                $self->fail('Unexpected node URI "' . $nodeUri . '"');
+            }
+            return 'http://www.example.org/some-node-link';
+        }));
```

Comment:
wait how is this testing that `createNodeUri` is called with 'html' when it's mocking the `resolveNodeUri` call? as far as I understand this test, it doesn't actually test the changed behavior..

does it fail if you remove the actual change?

Reply:
Sorry for not replying to this for a long time. I need to recheck the changes i made then. Is the used format (always html) the prefered way or should there be an option to customize the format used?
However, i need some help with the tests when it comes to an update.